src/main/java/com/petrych/screenshotter/service/ScreenshotServiceImpl.java

```diff
@@ -104,26 +104,34 @@ public File getScreenshotFileById(Long id) {
 	// store
 	
 	@Override
-	public String storeFile(String urlString) throws MalformedURLException {
+	public Screenshot storeFile(String urlString) throws MalformedURLException {
 		
 		UrlUtil.isUrlValid(urlString);
 		
-		String fileName = new ScreenshotMaker(getStorageLocation()).createFromUrl(urlString);
+		String fileName = "";
+		boolean fileNameUnique = false;
+		while (!fileNameUnique) {
+			fileName = FileUtil.generateFileName();
+			fileNameUnique = !screenshotFileExists(fileName);
```

Comment:
Ooh, I think this carries a very heavy performance penalty.. Are you walking through all files in every iteration here? 😟